src/alluvium/consumer.py

```diff
@@ -0,0 +1,102 @@
+import json
+import logging
+from abc import ABC, abstractmethod, abstractproperty
+from typing import Type, TypeVar
+
+import boto3
+
+
+class CDCEventProcessor(ABC):
+    """Encasulates post processing logic depending on the cdc format you expect to receive from the database."""
+
+    def __init__(self, payload):
+        self.cdc_event = json.loads(payload)
+
+    @abstractproperty
+    def should_keep(self):
+        pass
+
+    @abstractmethod
+    def post_process(self):
+        pass
+
+
+class Wal2JsonProcessorV1(CDCEventProcessor):
+    """
+    Post processing logic for postgres WAL2JSON Format v1.
+
+    Documentation: https://access.crunchydata.com/documentation/wal2json/2.0/
+    """
+
+    def __init__(self, payload):
+        super(Wal2JsonProcessorV1, self).__init__(payload)
+
+    @property
+    def should_keep(self):
+        return bool(self.cdc_event["change"])
+
+    def post_process(self):
+        return self.cdc_event
+
+
+class Wal2JsonProcessorV2(CDCEventProcessor):
+    """
+    Post processing logic for postgres WAL2JSON Format v2.
+
+    Documentation: https://access.crunchydata.com/documentation/wal2json/2.0/
+    """
+
+    IGNORE_ACTIONS = {"B", "C"}
+
+    def __init__(self, payload):
+        super(Wal2JsonProcessorV2, self).__init__(payload)
+
+    @property
+    def should_keep(self):
+        return self.cdc_event["action"] not in self.IGNORE_ACTIONS
+
+    def post_process(self):
+        return self.cdc_event
+
+
+ProcessorType = TypeVar("CDCEventProcessor", bound=CDCEventProcessor)
+
+
+class Consumer(ABC):
+    """Defines the consumer abstract class used by psycopg2's consume_stream."""
```

Comment:
Call out for Andrew. The error boundary conditions are non existant atm. What requirements would you all have when putting this into production? Also what is your take on python3 parameter typing versus schematics?